zcash_client_sqlite/src/wallet.rs

```diff
@@ -3548,6 +3546,68 @@ pub(crate) fn store_decrypted_tx<P: consensus::Parameters>(
         )?;
     }
 
+    // For each transaction that spends an output of this transaction and does not already have a
+    // known fee value, set the fee if possible.
+    let mut spending_txs_stmt = conn.prepare(
+        "SELECT DISTINCT t.id_tx, t.raw, t.block, t.expiry_height
+         FROM transactions t
+         -- find transactions that spend transparent outputs of the decrypted tx
+         LEFT OUTER JOIN transparent_received_output_spends ts
+            ON ts.transaction_id = t.id_tx
+         LEFT OUTER JOIN transparent_received_outputs tro
+            ON tro.transaction_id = :transaction_id
+            AND tro.id = ts.transparent_received_output_id
+         -- find transactions that spend Sapling outputs of the decrypted tx
+         LEFT OUTER JOIN sapling_received_note_spends ss
+            ON ss.transaction_id = t.id_tx
+         LEFT OUTER JOIN sapling_received_notes srn
+            ON srn.tx = :transaction_id
+            AND srn.id = ss.sapling_received_note_id
+         -- find transactions that spend Orchard outputs of the decrypted tx
+         LEFT OUTER JOIN orchard_received_note_spends os
+            ON os.transaction_id = t.id_tx
+         LEFT OUTER JOIN orchard_received_notes orn
+            ON orn.tx = :transaction_id
+            AND orn.id = os.orchard_received_note_id
```

Comment:
Do these output cases matter? Having the full transaction that receives a shielded note should have no bearing on the ability to determine the fee of the downstream tx spending that note.

If there is a motivation for doing this for all downstreams (rather than only looking at transparent downstreams), please document it.